contracts/Origins/OriginsBase.sol

```diff
@@ -472,9 +611,53 @@ contract OriginsBase is IOrigins, OriginsEvents {
 	}
 
 	/**
-	 * @notice Internal Function to transfer the token during buying.
+	 * @notice Internal function to check the stake data based on the tier and user address.
+	 * @param _tierID The tier id to get the stake requirement details.
+	 * @param _userAddress The user address whose stake has to be checked.
+	 */
+	function _checkStakesByTier(uint256 _tierID, address _userAddress) internal view returns (bool) {
+		/// @notice Checking if user has enough stake.
+		Stake memory stakes = stakeCondition[_tierID];
+
+		uint256 userStake;
+		for (uint256 i = 0; i < stakes.date.length; i++) {
+			userStake = userStake.add(stakes.staking.getPriorWeightedStake(_userAddress, stakes.blockNumber[i], stakes.date[i]));
+		}
+		/// @notice Averaging out the stakes
+		userStake = userStake.div(stakes.date.length);
+
+		/// @notice Checking is user have have enough stake.
+		if (userStake < stakes.minStake) {
+			return false;
+		}
+		/// @notice Checking the user stake does not exceed any limit if maxStake is set.
+		if (stakes.maxStake > 0 && userStake > stakes.maxStake) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * @notice Internal Function to check if the user is verified for the particular tier.
+	 * @return true if the user is verified, false otherwise.
+	 */
+	function _checkVerification(Tier memory _tierDetails, uint256 _tierID) internal view returns (bool) {
+		/// @notice Checking if verification is set and if user has permission.
+		if (_tierDetails.verificationType == VerificationType.None) {
+			revert("OriginsBase: No one is allowed for sale.");
```

Comment:
consider returning false for view functions instead of reverting

Reply:
Solved in 41d4ef7